common/reconciliation/common/types.go

```diff
@@ -83,16 +83,21 @@ const (
 
 // The following are types related to Invariant.
 type (
-	// Execution is an execution which should be checked or fixed.
+	// Execution is a base type for executions which should be checked or fixed.
 	Execution struct {
-		ShardID     int
-		DomainID    string
-		WorkflowID  string
-		RunID       string
+		ShardID    int
+		DomainID   string
+		WorkflowID string
+		RunID      string
+		State      int
```

Comment:
Does Execution serve as a base type for both concrete and current if so do all these fields apply to current?  I don't think current execution specifies State?

Reply:
Yep, according to DB records  It looks like we do update workflow state in current execution.